mongo/errors.go

```diff
@@ -71,6 +73,60 @@ func replaceErrors(err error) error {
 	return err
 }
 
+// IsDuplicateKeyError returns true if err is a duplicate key error
+func IsDuplicateKeyError(err error) bool {
+	// handles SERVER-7164 and SERVER-11493
+	for ; err != nil; err = unwrap(err) {
```

Comment:
Just curious, when would errors be nested in a way that would require this `unwrap()` for-loop?

Reply:
Ah ok, interesting. Thanks!